Pull Request: Allow Customization of "mason/bin" Path Handling

TBytes404 opened the pull request:
## Pull Request: Allow Customization of "mason/bin" Path Handling

### Description
This pull request introduces a new feature that allows users to customize the handling of the "mason/bin" path in the system. The change is achieved by adding the `place_mason_bin` option to the [`custom/chadrc.lua` file][1]. Users can now choose whether to prepend or append the "mason/bin" path based on their specific needs.

I have set the default option to prepend, aligning with the existing setting used in [`mason.nvim`][2] for consistency.

This change was prompted by a [discussion][3] regarding the need for more flexibility in managing the "mason/bin" path.

### Request
I kindly request a thorough review and verification of this pull request. Your feedback and suggestions are highly appreciated. Thank you!

[1]:https://github.com/TirtharajPramanik/example_config/blob/d19060f029642f039604d2de12b91270c0e52720/chadrc.lua#L15C1-L21
[2]:https://github.com/williamboman/mason.nvim/blob/db58ec7bc9248fb1878172f90cadc825b77564a6/lua/mason/settings.lua#L10-L15
[3]:https://github.com/NvChad/NvChad/pull/2031#issuecomment-1606449443

TBytes404 committed:
conditionally prepend or append mason/bin to path with mason_path option

This change allows users to choose whether to prepend or append mason/bin to path, by setting mason_path option in custom/chadrc.lua file

adamrp commented:
This looks useful, thanks @TirtharajPramanik! Recommend also adding comments in `lua/plugins/config/mason.lua` as well as `lua/plugins/init.lua` indicating that the user should not attempt to set the `PATH` option via the plugin's settings since doing so will not play nicely with the explicit path manipulation in `lua/core/init.lua`

TBytes404 committed:
cleaned up code and added descriptive comments

Add configurable 'place_mason_bin' option in custom/chadrc.lua for flexible mason/bin path handling

TBytes404 reviewed (commented):
Enhanced code with descriptive comments to alert users about potential tradeoffs when customizing options
